doc/contributing/adding-v8-fast-api.md

```diff
@@ -0,0 +1,147 @@
+
+Example of a JavaScript:
+
+```javascript
+// Let divide be a function that provides fast api calls.
+const { divide } = internalBinding('custom_namespace');
+
+function divide(a, b) {
+  return divide(a, b, { fallback: a === 0 })
+}
+```
+
+* C++ land
+
+Example of a conditional fast path on C++
+
+```c++
+// Anywhere in the execution flow, you can set fallback and stop the execution.
+static double divide(const int32_t a,
+                     const int32_t b,
+                     v8::FastApiCallbackOptions& options) {
+  if (b == 0) {
+    options.fallback = true;
+  } else {
+    return a / b;
+  }
+}
+```
+
+## Example
+
+A typical function that communicates between JavaScript and C++ is as follows.
+
+* On the JavaScript side:
+
+```javascript
+const { divide } = internalBinding('custom_namespace');
+```
+
+* On the C++ side:
+
+```c++
+#include "v8-fast-api-calls.h"
+
+namespace node {
+namespace custom_namespace {
+
+static void divide(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args[0]->IsInt32());
+  CHECK(args[1]->IsInt32());
+  auto a = args[0].As<v8::Int32>();
+  auto b = args[1].As<v8::Int32>();
+
+  if (b->Value() == 0) {
+    return node::THROW_ERR_INVALID_STATE(env, "Error");
+  }
+
+  double result = a->Value() / b->Value();
+  args.GetReturnValue().Set(result);
+}
+
+static double FastDivide(const int32_t a,
+                         const int32_t b,
+                         v8::FastApiCallbackOptions& options) {
+  if (b == 0) {
+    options.fallback = true;
+  } else {
+    return a / b;
+  }
+}
+
+CFunction fast_divide_(CFunction::Make(FastDivide));
+
+static void Initialize(Local<Object> target,
+                       Local<Value> unused,
+                       Local<Context> context,
+                       void* priv) {
+  SetFastMethod(context, target, "divide", Divide, &fast_divide_);
+}
+
+void RegisterExternalReferences(ExternalReferenceRegistry* registry) {
+  registry->Register(Divide);
+  registry->Register(FastDivide);
+  registry->Register(fast_divide_.GetTypeInfo());
+}
+
+} // namespace custom_namespace
+} // namespace node
+
+NODE_BINDING_CONTEXT_AWARE_INTERNAL(custom_namespace,
+                                    node::custom_namespace::Initialize);
+NODE_BINDING_EXTERNAL_REFERENCE(
+                      custom_namespace,
+                      node::custom_namespace::RegisterExternalReferences);
+```
+
+* Update external references ([`node_external_reference.h`](/src/node_external_reference.h))
+
+Since our implementation used
+`double(const int32_t a, const int32_t b, v8::FastApiCallbackOptions& options)`
+signature, we need to add it to external references.
+
+Example declaration:
+
+```c++
+using CFunctionCallbackReturningDouble = double (*)(const int32_t a,
+                                                    const int32_t b,
+                                                    v8::FastApiCallbackOptions& options);
+```
```

Comment:
````suggestion
  
  Since our implementation used
  `double(const int32_t a, const int32_t b, v8::FastApiCallbackOptions& options)`
  signature, we need to add it to external references.
  
  Example declaration:
  
  ```cpp
  using CFunctionCallbackReturningDouble = double (*)(const int32_t a,
                                                      const int32_t b,
                                                      v8::FastApiCallbackOptions& options);
  ```
````